Mathlib/Combinatorics/Digraph/Basic.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2024 Kyle Miller, Jack Cheverton. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Jack Cheverton
+-/
+import Mathlib.Order.CompleteBooleanAlgebra
+import Mathlib.Data.Fintype.Pi
+import Mathlib.Tactic.ApplyFun
+
+/-!
+# Digraphs
+
+This module defines directed graphs on a vertex type `V`.
+
+## Main definitions
+
+* `Digraph` is a structure for the relation
+
+* `CompleteAtomicBooleanAlgebra` instance: Under the subgraph relation, `Digraph` forms a
+  `CompleteAtomicBooleanAlgebra`. In other words, this is the complete lattice of spanning subgraphs
```

Comment:
Really "spanning" here? The empty digra

Reply:
It does because it's only empty in the sense of having no edges — it still has all the vertices `V`.

Reply:
Oops, misled by the fact that spanning trees are connected by virtue of being trees, I was thinking that a spanning subgraph should be connected. 

Sorry for the noise!